pkg/streamingpromql/testdata/upstream/aggregators.test

```diff
@@ -199,14 +199,12 @@ eval instant at 50m ceil(0.004 * http_requests{group="production",job="api-serve
 # 	{group="production", instance="1", job="api-server"} 10
 
 # Standard deviation and variance.
-# Unsupported by streaming engine.
-# eval instant at 50m stddev(http_requests)
-#   {} 229.12878474779
+eval instant at 50m stddev(http_requests)
+  {} 229.12878474779
 
-# Unsupported by streaming engine.
-# eval instant at 50m stddev by (instance)(http_requests)
-#   {instance="0"} 223.60679774998
-#   {instance="1"} 223.60679774998
+eval instant at 50m stddev by (instance)(http_requests)
+  {instance="0"} 223.60679774998
+  {instance="1"} 223.60679774998
```

Comment:
Could you add a test case of our own that has multiple groups that have different output values?

It'd also be good to see a test with NaN and infinite input values.

Reply:
Fair point to call out cases for infinity. The stddev of anything with infinity should be infinity. However, Prometheus returns NaN (and so do we).

Reply:
>The stddev of anything with infinity should be infinity. However, Prometheus returns NaN (and so do we).

That seems wrong - is it worth raising an issue with Prometheus to confirm the expected behaviour here?

Reply:
For now I've extended the change in Prometheus to handle infinities consistently. With that change, and MQE, any stddev containing infinity is NaN (or undefined). https://github.com/prometheus/prometheus/pull/14941